api/osrd_infra/migrations/__init__.py

```diff
@@ -2,10 +2,10 @@
 from django.db import migrations, models
 
 
-def run_sql_add_foreign_key(model_name: str, field_name: str, link_model: str):
+def run_sql_add_foreign_key(model_name: str, field_name: str, link_model: str, nullable: bool = False):
     return migrations.RunSQL(
         f"""ALTER TABLE osrd_infra_{model_name}
-                ADD {field_name}_id INTEGER,
+                ADD {field_name}_id INTEGER {"NULL" * nullable},
```

Comment:
Using a ternary condition looks better.

```suggestion
                ADD {field_name}_id INTEGER {"NULL" if nullable else ""},
```